install/kubernetes/cilium/templates/cilium-operator-deployment.yaml

```diff
@@ -205,8 +205,13 @@ spec:
 {{- if and (or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (ge .Capabilities.KubeVersion.Minor "17") (gt .Capabilities.KubeVersion.Major "1")) .Values.enableCriticalPriorityClass }}
       priorityClassName: system-cluster-critical
 {{- end }}
+{{- if .Values.serviceAccounts.operator.create }}
       serviceAccount: cilium-operator
       serviceAccountName: cilium-operator
+{{- else }}
+      serviceAccount: {{ .Values.operator.serviceAccountName }}
+      serviceAccountName: {{ .Values.operator.serviceAccountName }}
```

Comment:
nit:
```suggestion
      serviceAccount: {{ .Values.operator.serviceAccountName | quote }}
      serviceAccountName: {{ .Values.operator.serviceAccountName | quote }}
```

Reply:
@youssefazrak yes along some other YAML basic values (e.g. `yes`, `False`). I'm not 100% sure on the limitations of service account names in k8s, but better be safe than sorry.

Reply:
I followed the below principle as mentioned [here](https://v2.helm.sh/docs/chart_best_practices/#make-types-clear). 

> The easiest way to avoid type conversion errors is to be explicit about strings, and implicit about everything else. Or, in short, quote all strings.